rust-runtime/aws-smithy-types/src/instant/mod.rs

```diff
@@ -212,10 +187,28 @@ impl Instant {
         ))
     }
 
-    /// Converts number of milliseconds since the Unix epoch into an `Instant`.
-    pub fn from_epoch_millis(epoch_millis: i64) -> Instant {
-        let (seconds, millis) = div_mod_floor(epoch_millis, MILLIS_PER_SECOND);
-        Instant::from_secs_and_nanos(seconds, millis as u32 * NANOS_PER_MILLI)
+    /// Read 1 date of `format` from `s`, expecting either `delim` or EOF
+    ///
+    /// Enable parsing multiple dates from the same string
+    pub fn read(s: &str, format: Format, delim: char) -> Result<(Self, &str), InstantParseError> {
```

Comment:
we should probably fuzz this if we haven't already since it's another entrypoint for untrusted input